### PR TITLE
Add SupportedDestinationSyncModes to destination specs objects

### LINKED ISSUE #2603: Add DestinationSupportedSyncMode to API
## Tell us about the problem you're trying to solve
AirbyteCatalog declares `supportedSyncMode`, we should have one for destination sync mode too from #2370 

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Closes #2603 

## How
- Add `supported_destination_sync_modes` in destination connectors `specs.json`
- Add `SupportedDestinationSyncModes` to `DestinationDefinitionSpecificationRead`

## Pre-merge Checklist
- [x] *Run integration tests*
- [x] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

**Comment by ChristopheDuong** [comment]
Hi @Jamakase , is it enough for the Front-end if the field `SupportedDestinationSyncModes` is added to `DestinationDefinitionSpecificationRead`?

On the connection page, it should be then possible to retrieve the capabilities of the selected destination and verify that It supports one of the sync modes such as append, overwrite, or append_dedup and tweak the UI accordingly?

WDYT? 

Thanks!

**Comment by ChristopheDuong** [comment]
> @ChristopheDuong to support it on the frontend it should be implemented the same way we have `syncMode` for sources: it is in `AirbyteStream` object. So I expect `supportedDestinationSyncModes` to be somewhere on the same level ( in stream ). As it is possible to have different supported sync modes for different streams - then it means it can't be destination wide.

The supported sync modes from sources are indeed mixed and depending on the stream, some can be incremental, and some can be full-refresh, or both.

The supported destination sync mode is actually destination wide. It is not specific to streams and is not dependent on streams at all. But it depends on the destination that you have chosen for the connection/sync that you are setting up. 

However, in the end, you do want to configure the destination sync mode at a stream level because the final configuration is a combination of source and destination sync modes.

**Comment by ChristopheDuong** [comment]
> @ChristopheDuong to support it on the frontend it should be implemented the same way we have `syncMode` for sources: it is in `AirbyteStream` object. So I expect `supportedDestinationSyncModes` to be somewhere on the same level ( in stream ). 

In the backend, we "produce" the `AirbyteCatalog` that contains `AirbyteStream` after a call to `discoverSchema` on the source docker image. 

In what you are describing, I would have to add a call to `getSpec` to the destination connector right after the `discoverSchema` of the source, so I can pre-fill the catalog with the supported destination sync modes from the destination before creating the connection in the FE.

Is that correct @cgardens ?

**Comment by ChristopheDuong** [comment]
Artem's answer on slack with this iteration of changes to the Back-End:

> ok then. If its not stream specific - lets have it from destinationSpecification entity

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -1583,6 +1583,10 @@ components:
           $ref: "#/components/schemas/DestinationDefinitionSpecification"
         jobInfo:
           $ref: "#/components/schemas/SynchronousJobRead"
+        supportedDestinationSyncModes:
```
Comment (sherifnada) [inline airbyte-api/src/main/openapi/config.yaml:1586]: is the `destination` here extraneous?

**Review by sherifnada — APPROVED** [review]

File: airbyte-api/src/main/openapi/config.yaml (hunk 1 above)
Comment (ChristopheDuong) [reply airbyte-api/src/main/openapi/config.yaml:1586]: it is (i actually left it out at first), but then I thought we could confuse it with the source syncModes if it's called `supportedSyncModes` so I re-added the destination back in the name...

**Pull request merged**